pic32/cores/pic32/WInterrupts.c

```diff
@@ -255,6 +256,7 @@ void detachInterrupt(uint8_t interruptNum)
     }
 }
 
+#define VEC_ENT_CODE(I,B,P) int c = intmode[I]; IFS0bits.B=0; if (intFunc[I] != 0) (*intFunc[I])(); if(c < CHANGE && c == digitalRead(P)) { IFS0bits.B=1; }
```

Comment:
It would be preferable to use direct port access rather than digitalRead for the level check. digitalRead is slow and adds lots of overhead to an interrupt that is supposed to be rapid-response. It would be better to have an array of pin information (pointer to the IO structure and the pin mask) to directly access the registers.